src/lightning_lite/plugins/environments/slurm_environment.py

```diff
@@ -134,7 +128,17 @@ def local_rank(self) -> int:
     def node_rank(self) -> int:
         return int(os.environ["SLURM_NODEID"])
 
-    def resolve_root_node_address(self, root_node: str) -> str:
+    @staticmethod
+    def resolve_root_node_address(nodes: str) -> str:
+        """The node selection format in SLURM supports several formats.
+
+        This function selects the first host name from
+
+        - a space-separated list of host names, e.g., 'host0 host1 host3' yields 'host0' as the root
+        - a comma-separated list of host names, e.g., 'host0,host1,host3' yields 'host0' as the root
+        - the range notation with brackets, e.g., 'host[5-9]' yields 'host5' as the root
+        """
+        root_node = nodes.split(" ")[0].split(",")[0]
```

Comment:
If you prefer, this function can be made simpler with a single regex:

    nodes = re.sub("\[(.*?)[,-].*?\]", "\\1", nodes)  # Take the first node of every node range
    return nodes.split(" ")[0].split(",")[0]

Reply:
Yeah that's awesome, I integrated your suggestion